PSReadLine/BasicEditing.cs

```diff
@@ -246,6 +246,74 @@ public static void DeleteCharOrExit(ConsoleKeyInfo? key = null, object arg = nul
             _singleton.DeleteCharImpl(1, orExit: true);
         }
 
+        /// <summary>
+        /// Upcase the current word and move to the next one.
+        /// </summary>
+        public static void UpcaseWord(ConsoleKeyInfo? key = null, object arg = null)
+        {
+            if (_singleton._current >= _singleton._buffer.Length)
+            {
+                Ding();
+                return;
+            }
+
+            int wordlen = _singleton.FindForwardWordPoint(_singleton.Options.WordDelimiters) - _singleton._current;
+            string word = _singleton._buffer.ToString(_singleton._current, wordlen);
+            word = word.ToUpper();
+
+            _singleton._buffer.Remove(_singleton._current, wordlen);
+            _singleton._buffer.Insert(_singleton._current, word);
+
+            _singleton.MoveCursor(Math.Min(_singleton._current + wordlen, _singleton._buffer.Length));
+            _singleton.Render();
+        }
+
+        /// <summary>
+        /// Downcase the current word and move to the next one.
+        /// </summary>
+        public static void DowncaseWord(ConsoleKeyInfo? key = null, object arg = null)
+        {
+            if (_singleton._current >= _singleton._buffer.Length)
+            {
+                Ding();
+                return;
+            }
+
+            int wordlen = _singleton.FindForwardWordPoint(_singleton.Options.WordDelimiters) - _singleton._current;
+            string word = _singleton._buffer.ToString(_singleton._current, wordlen);
+            word = word.ToLower();
+
+            _singleton._buffer.Remove(_singleton._current, wordlen);
+            _singleton._buffer.Insert(_singleton._current, word);
+
+            _singleton.MoveCursor(Math.Min(_singleton._current + wordlen, _singleton._buffer.Length));
+            _singleton.Render();
+        }
+
+        /// <summary>
+        /// Capitalize the current word and move to the next one.
+        /// </summary>
+        public static void CapitalizeWord(ConsoleKeyInfo? key = null, object arg = null)
+        {
+            if (_singleton._current >= _singleton._buffer.Length)
+            {
+                Ding();
+                return;
+            }
+
+            int wordlen = _singleton.FindForwardWordPoint(_singleton.Options.WordDelimiters) - _singleton._current;
+
+            char[] word = _singleton._buffer.ToString(_singleton._current, wordlen).ToLower().ToCharArray();
+            int idxFirstNonWhitespace = Array.FindIndex(word, x => !char.IsWhiteSpace(x));
```

Comment:
> x => !char.IsWhiteSpace(x)

This doesn't work for characters that is not white space and not word letters. An example is `abc *** fff`. Move cursor to right before the first `*`, and press `alt+c`. The input is not changed with this change in PowerShell, but will be comes `abc *** Fff` in bash.